### DISCUSSION TITLE
ApiClient autogenerated request signing

### DISCUSSION THREAD

**Discussion by smargoli2** [abuzuhri/Amazon-SP-API-CSharp#303] [Q&A] [answered]
Hi,
The [selling partner api models](https://github.com/amzn/selling-partner-api-models/tree/main/models) have an API client that looks like it signs the requests itself; there is no need to use the lwaauthorizationsigner and awssigv4signer; it is not done automatically. AM I understanding this correctly?
How would we implement one client for all the models?

**Comment by abuzuhri** [comment, marked as answer]
yes library do that automatically
